pointcloud/basic_octree_utils.go

```diff
@@ -165,11 +161,38 @@ func (octree *basicOctree) helperIterate(lowerBound, upperBound, idx int, fn fun
 			idx += numPoints
 		}
 
-	case LeafNodeFilled:
+	case leafNodeFilled:
 		ok = fn(octree.node.point.P, octree.node.point.D)
 
-	case LeafNodeEmpty:
+	case leafNodeEmpty:
 	}
 
 	return ok
 }
+
+// ConvertPointCloudToBasicOctree converts a pointcloud into a octree pointcloud representation.
+func ConvertPointCloudToBasicOctree(cloud PointCloud) (PointCloud, error) {
+	center := r3.Vector{
+		X: cloud.MetaData().MinX + (cloud.MetaData().MaxX-cloud.MetaData().MinX)/2,
+		Y: cloud.MetaData().MinY + (cloud.MetaData().MaxY-cloud.MetaData().MinY)/2,
+		Z: cloud.MetaData().MinZ + (cloud.MetaData().MaxZ-cloud.MetaData().MinZ)/2,
+	}
+
+	side := math.Max((cloud.MetaData().MaxX-cloud.MetaData().MinX),
+		math.Max((cloud.MetaData().MaxY-cloud.MetaData().MinY),
+			(cloud.MetaData().MaxZ-cloud.MetaData().MinZ))) * 1.01
+
+	basicOctPC, err := NewBasicOctree(center, side)
+	if err != nil {
+		return &BasicOctree{}, err
```

Comment:
For my own understanding, why can't we return `nil, err` here?

Reply:
We can! Not sure why I originally had to set it to `&BasicOctree{}` but its fine to return a nil